app/main.py

```diff
@@ -103,11 +103,11 @@ def init_primaite():
             env_view.empty()
 
             # Run simulation
-            action, info = agent._calculate_action_info(obs)
+            action, prompt, reasoning = agent._calculate_action_info(obs)
```

Comment:
i'm not sure I like that we have an agent return the prompt and reasoning as well irrespective of its type. ideally here we would check if the agent is an llm agent, in which case we would get the prompt and reasoning. but it's fine for now, we can come back to this later

Reply:
Updated see next commit